tactic/interactive.lean

```diff
@@ -474,5 +474,18 @@ meta def apply_field : tactic unit :=
 propagate_tags $
 get_current_field >>= applyc
 
+/-- `elim_cast e with x` matches on `cast _ e` in the goal and replaces it with
+    `x`. It also adds `Hx : e == x` as an assumption.
+
+    `elim_cast! e with x` acts similarly but reverts `Hx`.
+-/
```

Comment:
What if there are multiple casts on `e`? i.e. `cast h1 e` and `cast h2 e` of different types. Do they both get names?

Reply:
Only one of them gets replaced. In short, each call to `elim_cast` introduces only one new variable. It will replace all occurrences of a given expression being cast to the same type.

Do you think the behaviour should be different?

I'll add those details to the example and the header comment.

Reply:
I could add the `elim_cast*` option to introduce as many new variables as necessary to eliminate all the cast expressions of a given expression.

Reply:
I think doing them one at a time is fine, it's probably a rare use case anyway. Also, this tactic feels incomplete without support for `eq.rec`, `eq.rec_on`, `eq.subst`, `eq.substr`, `eq.mp` and `eq.mpr` as well, turning all of them into heq assumptions. All of them are defeq to `eq.rec` of something, so your tactic should probably work from there.

Reply:
Ok, I think this will need some consideration. I'll look into it.

Reply:
By the way, the analogue of your cast theorems already exists for `eq.rec_on`: `heq_of_eq_rec_left`, `heq_of_eq_rec_right`, and `eq_rec_heq` should do the trick. (There is also `cast_heq` which you may have noticed.)